rigs/heft-web-rig/shared/webpack-base.config.js

```diff
@@ -0,0 +1,261 @@
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
+const { BundleAnalyzerPlugin } = require('webpack-bundle-analyzer');
+const { merge, mergeWithCustomize, mergeWithRules, unique } = require('webpack-merge');
+const sass = require('sass');
+const autoprefixer = require('autoprefixer');
+
+/**
+ * If the "--production" command-line parameter is specified when invoking Heft, then the
+ * "production" function parameter will be true.  You can use this to enable bundling optimizations.
+ */
+function createWebpackConfig({ env, argv, projectRoot, configOverride }) {
+  const { production } = env;
+
+  const defaultArgs = {
+    // Documentation: https://webpack.js.org/configuration/mode/
+    mode: production ? 'production' : 'development',
+    resolve: {
+      extensions: ['.mjs', '.js', '.json']
+    },
+    output: production
+      ? {
+          chunkFilename: '[name].[contenthash].js',
+          filename: '[name].[contenthash].js',
+          sourceMapFilename: '[name].[contenthash].js.map'
+        }
+      : {},
+    module: {
+      rules: [
+        {
+          // The source-map-loader extracts existing source maps from all JavaScript entries. This includes both
+          // inline source maps as well as those linked via URL. All source map data is passed to Webpack for
+          // processing as per a chosen source map style specified by the devtool option in webpack.config.js.
+          // https://www.npmjs.com/package/source-map-loader
+          test: /\.js$/,
+          include: [path.resolve(projectRoot, 'lib')],
+          enforce: 'pre',
+          use: [
+            {
+              loader: require.resolve('source-map-loader')
+            }
+          ]
+        },
+
+        {
+          // Recommendation for CSS styles:
+          //
+          // We recommend the newer .scss file format because its syntax is a proper superset of familiar CSS.
+          // The older .sass syntax is accepted for backwards compatibility.
+          //
+          // Handling of formats:
+          // - *.scss:          sass, autoprefixer, modules  <--- recommended
+          // - *.global.scss:   sass, autoprefixer, NO modules
+          // - *.css:           NO sass, NO autoprefixer, NO modules
+          //
+          // If someone wants CSS with modules, they can simply use the .sass file extension, since SASS
+          // is essentially a superset of CSS. (There's a small performance penalty for applying SASS to
+          // a plain CSS file, but the extra syntax validation probably justifies that cost.)
+          //
+          // The SASS docs are here: https://sass-lang.com/documentation/syntax
+          test: /\.(scss|sass|css)$/,
+          exclude: /node_modules/,
+          use: [
+            // "For production builds it's recommended to extract the CSS from your bundle being able to
+            // use parallel loading of CSS/JS resources later on. This can be achieved by using the
```

Comment:
Should this maybe be
```suggestion
            // "For production builds it's recommended to extract the CSS from your bundle so that you are
            // able to use parallel loading of CSS/JS resources later on. This can be achieved by using the
```
or some such? I'm not sure how to parse it as it is.

Reply:
The optimization is really less about parallelization and more that it lets you use the browser's optimized `<link rel="stylesheet" href="..." />` hot path for importing your CSS. Also improves caching.

Reply:
The quotation marks indicate that it is an excerpt from the Webpack website.  The main point was that Webpack recommends to do this.